.github/workflows/nighthawk.yml

```diff
@@ -0,0 +1,80 @@
+name: Nighthawk/Test CI/CD
+
+permissions:
+  contents: read
+on:
+  pull_request:
+  push:
+    branches:
+    - main
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+
+jobs:
+  check:
+    permissions:
+      contents: read
+      packages: read
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+        - build
+        - check_format
+    uses: ./.github/workflows/_ci.yml
+    with:
+      task: ${{ matrix.target }}
+
+  test:
+    permissions:
+      contents: read
+      packages: read
+    needs: check
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+        - benchmark_with_own_binaries
+        - test
+        - test_gcc
+    uses: ./.github/workflows/_ci.yml
+    with:
+      task: ${{ matrix.target }}
+
+  sanitizers:
+    permissions:
+      contents: read
+      packages: read
+    needs: test
```

Comment:
@mum4k i would just fold these into the test - it will speed everything up no end

Reply:
+1 if we have the resources to run these together, it will be much better.